### PR TITLE
Parse templates from configmaps

### PR DESCRIPTION
**Pull request by dinosk**
Adds func `ParseConfigMap` to parse a ConfigMap and return a map of CAPITemplates indexed by their name.
The name of the template is set to the key of the ConfigMap.Data map.

### CODE REVIEW

File: pkg/flavours/parse_test.go [hunk 1]
```diff
@@ -82,6 +85,51 @@ func TestParseFileFromFS_with_unknown_file(t *testing.T) {
...
+			Params: []TemplateParam{
+				{
+					Name:        "CLUSTER_NAME",
+					Description: "This is used for the cluster naming.",
+				},
+			},
+			ResourceTemplates: []CAPIResourceTemplate{},
+		},
+	}
+	if diff := cmp.Diff(want, tm["template1"], cmpopts.IgnoreFields(CAPITemplateSpec{}, "ResourceTemplates")); diff != "" {
```
Comment (bigkevmcd) [inline pkg/flavours/parse_test.go]: You could change this to be...

```go
want := map[string]*CAPITemplate{
    "template1": &CAPITemplate{
        TypeMeta: ...
    }
```

File: pkg/flavours/parse_test.go [hunk 2]
```diff
@@ -82,6 +85,51 @@ func TestParseFileFromFS_with_unknown_file(t *testing.T) {
...
 }
 
+
+func TestParseConfigMap(t *testing.T) {
+	cmBytes, err := ioutil.ReadFile("testdata/configmap1.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	decode := scheme.Codecs.UniversalDeserializer().Decode
```
Comment (bigkevmcd) [inline pkg/flavours/parse_test.go]: I'd simplify this to just `obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(cmBytes, nil, nil)`

Comment (dinosk) [reply pkg/flavours/parse_test.go]: Good point, thanks

**Review by bigkevmcd — APPROVED** [review]
Thanks for this!

**Pull request merged**